modal/cls.py

```diff
@@ -40,16 +42,45 @@
     import modal.app
 
 
+def _use_annotation_parameters(user_cls) -> bool:
+    has_parameters = any(isinstance(cls_member, _Parameter) for cls_member in user_cls.__dict__.values())
+    has_explicit_constructor = user_cls.__init__ != object.__init__
+    return has_parameters and not has_explicit_constructor
```

Comment:
We do that in `validate_construction_mechanism` which runs before this. At this point it will return False if both are declared, which I think is ok as the "lower level" behavior (works similar to data classes with explicit constructors)